### DISCUSSION TITLE
Dynamically Loaded Components cannot work.

### DISCUSSION THREAD

**Discussion by richardhahahaha** [idom-team/idom#1046] [Problem] [answered]
sample code is follow:

```
from reactpy import component, run, web

mui = web.module_from_template(
    "react@^17.0.0",
    "@material-ui/core@4.12.4",
    fallback="⌛",
)
Button = web.export(mui, "Button")

@component
def HelloWorld():
    return Button({"color": "primary", "variant": "contained"}, "Hello World!")

run(HelloWorld)
```

error info in browser console:

```
Failed to load module script: Expected a JavaScript module script but the server responded with a MIME type of "text/plain". Strict MIME type checking is enforced for module scripts per HTML spec.
```

**Comment by rmorshea** [comment]
Are you on Windows?

**Reply by Archmonger** [reply]
Also, what backend are you using?

**Reply by rmorshea** [reply]
^ That information should be shown in the logs when running with `reactpy.run`

**Comment by fsmosca** [comment]
I have the same error message. I use windows 10.

What works for me currently is to change the port.

```python
run(HelloWorld, port=7000)
```

port `8000`, does not work.

I also use this mui, I get it from [youtube](https://www.youtube.com/watch?v=JOqf-V1d0KY).

```python
mui = web.module_from_template(
    "react",
    "@mui/material",
    fallback="⌛",
)
```

### Sample counter

![image](https://github.com/reactive-python/reactpy/assets/22366935/06b5b977-b463-449e-8a32-388b9d269e85)

**Reply by Archmonger** [reply, marked as answer]
`module_from_template` is currently deprecated due to bugs and instability. We will eventually release an alternative to this, but for we recommend using our [custom JavaScript components API](https://reactpy.dev/docs/guides/escape-hatches/javascript-components.html#custom-javascript-components).
